### DISCUSSION TITLE
Destroy/recreate vs reset/reuse the instance

### DISCUSSION THREAD

**Discussion by obenno** [flekschas/regl-scatterplot#191] [Q&A] [answered]
Hi, 

thanks a lot for the wonderful library!

I'm working on a project that involves creating multiple instances of regl-scatterplot. I'm wondering about the best practice for handling these instances, particularly in scenarios where I need to frequently update or change the scatterplots. Is it okay to destroy and recreate regl-scatterplot instances multiple times within a function or it is better to reset the previous instance and reuse it instead?

Any insights or best practices you can share would be greatly appreciated. Thanks in advance for your help!

Bests,
oben

**Comment by flekschas** [comment, marked as answer]
I haven't done any specific performance testing but generally, destroying and recreating is more expensive. So I would probably start with resetting the existing plots and if that doesn't work for some reason, I would go for recreating plots.

**Reply by obenno** [reply]
Thanks a lot for the suggestion!
